pkg/controller/generic_reconciler_test.go

```diff
@@ -1046,3 +1047,102 @@ func createTestReconciler(scheme *runtime.Scheme, gvks []schema.GroupVersionKind
 	testGenericReconciler.resourceGVKs = gvks
 	return testGenericReconciler, nil
 }
+
+// BenchmarkGroupAppObjects measures the performance of grouping Kubernetes objects based on their labels.
+// The benchmark focuses on a scenario where a Reconciler needs to group different objects based on the 'app' label.
+// # Benchmark configuration:
+//
+// The benchmark is configured with the following parameters:
+// - namespace: "test" - The namespace in which the objects will be created.
+// - deploymentNumber: 100 - The number of Deployment objects to create to test.
+// - channelBuffer: 10 - The buffer size for the channel used in the grouping process.
+//
+// # How to run:
+//
+// To run the benchmark, run the following command:
+//
+//	go test -bench=BenchmarkGroupAppObjects
+//
+// Adjust the number of deployments or the buffer to match a given scenario.
+// Note that the local benchmark can be modified by other processes running in the background.
+// If the 'groupAppObjects' function is modified, run some benchmarks with before and after status
+// to compare performance improvements.
+func BenchmarkGroupAppObjects(b *testing.B) {
+	namespace := "test"
+	deploymentNumber := 100
+	channelBuffer := 10
+
+	// Given
+	objs := []client.Object{
+		&policyv1.PodDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-pdb-A-B-C", Namespace: namespace},
+			Spec: policyv1.PodDisruptionBudgetSpec{
+				Selector: &metav1.LabelSelector{
+					MatchExpressions: []metav1.LabelSelectorRequirement{
+						{
+							Key: "app", Operator: metav1.LabelSelectorOpIn,
+							Values: []string{"A", "B", "C"},
```

Comment:
I think this PDB never matches any deployment below, because the deployments have labels as `App-A`, `App-B`, etc.